### PR TITLE
created a simple calendar

### PR DESCRIPTION
**Pull request by Evgenb000**
- [DEMO LINK](https://evgenb000.github.io/layout_calendar/)
- [TEST REPORT LINK](https://evgenb000.github.io/layout_calendar/report/html_report/)

### CODE REVIEW

File: src/styles/calendar.scss [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  gap: $cells-gap;
+  max-width: $calendar-max-width;
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: flex-start;
+
+  margin: 50px auto;
```
Comment (yuliia-anisimova) [inline src/styles/calendar.scss]: You can't set margins for BEM-block.

File: src/styles/calendar.scss [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  @for $last-day from 28 through 31 {
+    &--month-length_#{$last-day}
+    &__day:nth-child(n + #{$last-day + 1}) {
+      display: none;
+    }
+  }
+
+  @each $name, $num in $calendar-day {
```
Comment (yuliia-anisimova) [inline src/styles/calendar.scss:48]: Maybe, it's better to create another file for @for and @each loop, but I'm not sure if it's necessary for your code

File: src/styles/calendar.scss [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      &:nth-child(#{$day}):before {
+        content: "#{$day}";
+      }
+    }
+  }
+
+  &__day:hover {
+    background-color: rgb(255, 107, 187);
+    transform: translate(0, -20px);
+  }
```
Comment (yuliia-anisimova) [inline src/styles/calendar.scss]:
```suggestion
  &__day {
    @include square($cell-size);

    background-color: #eee;
    border: 1px solid black;
    box-sizing: border-box;

    display: flex;
    align-items: center;
    justify-content: center;

    cursor: pointer;
    transition: background-color 0.5s, transform 0.5s;
    
    &:hover {
    background-color: rgb(255, 107, 187);
    transform: translate(0, -20px);
    }

    @for $day from 1 through 31 {
      &:nth-child(#{$day}):before {
        content: "#{$day}";
      }
    }
  }

```

File: src/styles/calendar.scss [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    transition: background-color 0.5s, transform 0.5s;
+
+    @for $day from 1 through 31 {
+      &:nth-child(#{$day}):before {
+        content: "#{$day}";
+      }
+    }
+  }
+
+  &__day:hover {
```
Comment (yuliia-anisimova) [inline src/styles/calendar.scss]: It's better to nest your :hover inside the &__day

**Review by yuliia-anisimova — CHANGES_REQUESTED** [review]
Great job! But please, check my comments and make the necessary changes

File: src/styles/calendar.scss (hunk 1 above)
Comment (VinnieJ-2k20) [reply src/styles/calendar.scss]: Yep.

File: src/styles/calendar.scss (hunk 2 above)
Comment (VinnieJ-2k20) [reply src/styles/calendar.scss:48]: No, keep them here.

**Review by VinnieJ-2k20 — CHANGES_REQUESTED** [review]
Remove margin rules from BEM block, and that's it!

**Review by VinnieJ-2k20 — CHANGES_REQUESTED** [review]
Code looks OK, but DEMO link is not working. Redeploy the project.,

**Review by liliia-livandovskaya — APPROVED** [review]

**Pull request closed without merging**